core/libp2p/protocol/kademlia/kad.hpp

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_KADEMLIA_KAD_HPP
+#define KAGOME_KADEMLIA_KAD_HPP
+
+#include "libp2p/network/network.hpp"
+#include "libp2p/protocol/kademlia/config.hpp"
+#include "libp2p/protocol/kademlia/value_store.hpp"
+#include "libp2p/routing/content_routing.hpp"
+#include "libp2p/routing/peer_routing.hpp"
+
+namespace libp2p::protocol::kademlia {
+
+  /**
+   * @class Kad
+   *
+   * Entrypoint to a kademlia network.
+   */
```

Comment:
Please, elaborate more on what is Kademlia network. It is understandable, that it's an implementation of `PeerRouting`, but what is it exactly? Can be useful for those, who read the code for the first time

Reply:
This is not a "final" version of Kad class. 
I **will** add docs once Kad is completed and I completely understand everything :)